Fixed annotations

sreichel opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->

Fixed someannotations that code analysation had complained about:

1. Magento\Catalog\Model\ResourceModel\Product\Collection::addAttributeToFilter()

`array` is a valid input type for `addAttributeToFilter()`

2. Magento\Quote\Model\QuoteAddress::setCollectShippingRates()

Everywhere in code `setCollectShippingRates()` uses `true` as parameter, but annotation wants an integer.

3. Magento\Customer\Model\Address\AbstractAddress::getCountry()

`getCountryId()` and `getCountry()` returns a string, not an integer 

![m2-countryid](https://user-images.githubusercontent.com/5022236/47720346-f5d88380-dc4d-11e8-861b-10ae13ec538a.PNG)

4. `setData()` returns `\Magento\Framework\DataObject` instead of `$this`

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. n/a

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

sreichel committed:
Fixed annotations

- Magento\Catalog\Model\ResourceModel\Product\Collection::addAttributeToFilter()
- Magento\Quote\Model\QuoteAddress::setCollectShippingRates()

magento-engcom-team commented:
Hi @sreichel. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me $VERSION instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

magento-engcom-team commented:
Hi @VladimirZaets, thank you for the review.
[ENGCOM-3331](https://jira.corp.magento.com/browse/ENGCOM-3331) has been created to process this Pull Request

sreichel committed:
Fixed annotations

- Magento\Customer\Model\Address\AbstractAddress::getCountry()
- Magento\Customer\Model\Address\AbstractAddress::getCountryId()

sreichel committed:
Fixed annotations for setData()

setData() should return objects class - that inherits from
